src/harness/testcases/WINNF_FT_S_HBT_testcase.py

```diff
@@ -0,0 +1,135 @@
+  def setUp(self):
+    self._sas, self._sas_admin = sas.GetTestingSas()
+    self._sas_admin.Reset()
+
+  def tearDown(self):
+    pass
+
+  def getHeartBeatRequest(self,cbsd_ids, grant_ids, previous_transmit_expire_time):     
+    heartbeat_request = []
+    for cbsd_id, grant_id, transmit_expire_time in\
+     zip(cbsd_ids, grant_ids, previous_transmit_expire_time):
+        status = 'AUTHORIZED'
+        if(transmit_expire_time is None):
+            status = 'GRANTED'
+        elif datetime.strptime(transmit_expire_time,'%Y-%m-%dT%H:%M:%SZ') <=\
+         (datetime.utcnow() + timedelta(seconds= 2)):
+            time.sleep(2)
+            status = 'GRANTED'
+        heartbeat_request.append({
+            'cbsdId': cbsd_id,
+            'grantId': grant_id,
+            'operationState': status
+        })
+    return {'heartbeatRequest': heartbeat_request}
+   
+  @winnforum_testcase
+  def test_WINNF_FT_S_HBT_9(self):
+    """ Array request, of three cbsds,SAS terminates the Grant for the one
+    that impacts incumbent activity.
+
+    successful heartbeat for the grants of two cbsds that don't impact
+     the incumbent and failed with responseCode 500 for the third one.
+    """
+    # STEP 1
+    # register three devices
+    # load  CBSD data
+    device_a = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_a.json'))) 
+    device_c = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_c.json')))
+    device_e = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_e.json')))
+    registration_request = [device_a, device_c, device_e]
+
+    # load and set Grants data
+    grant_a = json.load(
+        open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_c = json.load(
+        open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_e = json.load(
+        open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_e['operationParam']['operationFrequencyRange']\
+                             ['lowFrequency'] = 3650000000
+    grant_e['operationParam']['operationFrequencyRange']\
+                             ['highFrequency'] = 3660000000
+    grant_request = [grant_a, grant_c, grant_e]
+    cbsd_ids, grant_ids = self.assertRegisteredAndGranted(registration_request,
+                                                           grant_request)
+    # First Heartbeat
+    request =  self.getHeartBeatRequest(cbsd_ids, grant_ids, [None, None, None])
+    response = self._sas.Heartbeat(request)['heartbeatResponse']
+    # check Heartbeat response 
+    self.assertEqual(len(first_hb_response), len(grant_ids))
+    for response_num, resp in enumerate(first_hb_response):
+        self.assertEqual(resp['cbsdId'], cbsd_ids[response_num])
+        self.assertEqual(resp['grantId'], grant_ids[response_num])
+        self.assertEqual(resp['response']['responseCode'], 0)
+        grant_expire_time = datetime.strptime(resp['grantExpireTime'],\
```

Comment:
I think we can avoid doing this and the following checks since they are not requested by the spec.